includes/class-admin.php

```diff
@@ -46,6 +53,35 @@ public static function admin_menu() {
 		}
 	}
 
+	/**
+	 * Collect admin menu notices about configuration problems or conflicts
+	 */
+	public static function get_admin_notices() {
+		$admin_notices = array();
+
+		$permalink_structure = \get_option( 'permalink_structure' );
+		if ( empty( $permalink_structure ) ) {
+			array_push( $admin_notices, 'You are using the ActivityPub plugin without setting a permalink structure.  This will prevent ActivityPub from working.  Please set a permalink structure.' );
```

Comment:
why not adding the admin notice directly here?

Reply:
Possibly over-engineering, but I think there are potentially many conflicts, environment problems, or misconfigurations that could be checked here.  They would all be displayed the same.  So it seems like repeating all of the display block for each one is a bit weird.  It's also an instinctive attempt to separate business logic from display logic.  If you prefer, I can certainly do it that way.

Reply:
I like the `get_admin_notices` function, but I would avoid the static `$admin_notices` var. So why not call `show_admin_notice( $admin_notice, 'error' )` to build the HTML directly here instead of pushing it to the array? this way we could add others later on?!?

This way we might also remove the `! \WP_Block_Type_Registry::get_instance()->is_registered( 'activitypub/followers' )` check?!?

Reply:
OK, sounds good.  I'll implement that when I get a chance.  Thanks for the review!

Reply:
thanks a lot!